projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/ai-assistant/with-ai-assistant.tsx

```diff
@@ -189,19 +189,22 @@ export const withAIAssistant = createHigherOrderComponent(
 
 			replaceBlock(
 				firstClientId,
-				transformToAIAssistantBlock( extendedBlockAttributes, blockType )
+				transformToAIAssistantBlock( blockType, extendedBlockAttributes )
 			);
 
 			removeBlocks( otherBlocksIds );
 		}, [ blocks, blockType, content, replaceBlock, clientIds, removeBlocks ] );
 
-		const rawContent = getRawTextFromHTML( props.attributes.content );
+		const blockControlProps = {
+			group: 'block',
+		};
+		const rawContent = getRawTextFromHTML( content );
```

Comment:
Are you running regexes, and setting innerHTML on every render of this block higher order component? https://github.com/Automattic/jetpack/pull/31901/files#diff-6dbc088debefb5e5c987ea76a69651132bf1c7d2234fd16ac3cbf0815c88ed20R95

Reply:
This higher order component is extending every block probably? So this is running on every render for every block in a post?

Reply:
Not every block, but pretty much every block https://github.com/Automattic/jetpack/blob/5daa8f1be69422fd0106a1b6a7fbb5c62105ed8c/projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/ai-assistant/index.ts#L20

Reply:
That is correct. Issues being addressed on https://github.com/Automattic/jetpack/pull/33681
ref p1697714798422019-slack-C02TQF5VAJD